src/components/ebay-phone-input/component.ts

```diff
@@ -0,0 +1,132 @@
+    countriesEnglish,
+} from "../../common/countries";
+import { AttrString } from "marko/tags-html";
+
+export interface CountryNames {
+    [index: string]: string;
+}
+
+export interface PhoneInputEvent {
+    originalEvent?: Event;
+    value?: string;
+    rawValue?: string;
+    callingCode?: string;
+    countryCode?: string;
+}
+
+interface PhoneInputInput extends Omit<MenuButtonInput, `on${string}`> {
+    disabled?: boolean;
+    "country-names"?: CountriesLanguageInterface;
+    "a11y-button-text"?: AttrString;
+    "a11y-menu-text"?: MenuButtonInput["a11yText"];
+    "floating-label"?: string;
+    readonly?: boolean;
+    invalid?: boolean;
+    "country-code"?: string;
+    value?: string;
+    "on-keyup"?: (event: PhoneInputEvent) => void;
+    "on-keypress"?: (event: PhoneInputEvent) => void;
+    "on-keydown"?: (event: PhoneInputEvent) => void;
+    "on-focus"?: (event: PhoneInputEvent) => void;
+    "on-blur"?: (event: PhoneInputEvent) => void;
+    "on-collapse"?: (event: PhoneInputEvent) => void;
+    "on-expand"?: (event: PhoneInputEvent) => void;
+    "on-change"?: (event: PhoneInputEvent) => void;
+}
+
+export interface Input extends WithNormalizedProps<PhoneInputInput> {}
+
+export interface State {
+    index: number;
+}
+
+class PhoneInput extends Marko.Component<Input, State> {
+    mask: any;
+
+    onCreate() {
+        this.state = {
+            index: 0,
+        };
+    }
+
+    get selectedCountry(): CountryInterface {
+        const countryNames = this.input.countryNames || countriesEnglish;
+        const currentCountryName = countryNames[this.state.index];
+        return countries[currentCountryName[1]];
+    }
+
+    onInput(input: Input) {
+        const countryNames: CountriesLanguageInterface =
+            input.countryNames || countriesEnglish;
+        const { countryCode } = input;
+        if (countryCode) {
+            let index = countryNames.findIndex(
+                (country) => country[1] === countryCode.toUpperCase(),
+            );
+            if (index === -1) {
+                index = 0;
+            }
+            this.state.index = index;
+        }
+    }
+    handleCountryChange(e: any) {
+        this.state.index = e.index;
+        // Trigger change event after state is updated so that the right mask value will be emitted
+        setTimeout(() => {
+            this.forwardEvent("change", { originalEvent: e });
+        }, 0);
+    }
+
+    forwardEvent(eventName: string, event: PhoneInputEvent) {
+        const selectedCountry = this.selectedCountry;
+
+        setTimeout(() => {
+            this.emit(eventName, {
+                originalEvent: event.originalEvent,
+                value: this.mask.value,
+                rawValue: stripNonDigits(this.mask.value || ""),
+                callingCode: selectedCountry.callingCode,
+                countryCode: selectedCountry.countryCode,
+            } satisfies PhoneInputEvent);
+        });
+    }
+
+    _initializeMask() {
```

Comment:
If you move event handling logic into this file instead of `mask` you'll be able to remove this function, `onMount`, and `onDestroy`. Up to you, but I think it's a cleaner solution

Reply:
Not sure what you mean. 
Still need to have those events. We just need to call them directly instead of an external file.

Reply:
Sure, but if you handle the events using Marko instead of attaching them directly with JavaScript you can be sure that they will be cleaned up properly and it is more obvious where they are coming form

Reply:
Resolving this for now, will use the other comment